spring-cloud-skipper-server/src/main/java/org/springframework/cloud/skipper/server/repository/ReleaseRepositoryImpl.java

```diff
@@ -86,4 +86,11 @@ public Release findByNameAndVersion(String releaseName, int version) {
 		}
 		return releases;
 	}
+
+	@Override
+	public Release findLatestReleaseIfDeleted(String releaseName) {
+		Release latestRelease = this.releaseRepository.findTopByNameOrderByVersionDesc(releaseName);
+		return (latestRelease != null &&
+				latestRelease.getInfo().getStatus().getStatusCode().equals(StatusCode.DELETED)) ? latestRelease : null;
```

Comment:
I wonder if we need to handle 'FAILED' as well.  We need to review the states in StatusCode....